### ISSUE TITLE
Docker rootless v25.0.2 error mounting "/etc/hosts" loki-docker-driver

### ISSUE THREAD

**Issue by JesseVS** [grafana/loki#11934]
**Describe the bug**
Since docker version 25.0.x the docker plugin grafana/loki-docker-driver does not start with the following error:

```
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/etc/hosts" to rootfs at "/etc/hosts": mount /etc/hosts:/etc/hosts (via /proc/self/fd/6), flags: 0x5021: operation not permitted: unknown
```

**To Reproduce**
1. Install Debian 11
2. Follow installation https://docs.docker.com/engine/install/debian/#install-using-the-repository
3. login as user
4. run dockerd-rootless-setuptool.sh install
5. run docker plugin install grafana/loki-docker-driver:latest --alias loki --grant-all-permissions

**Expected behavior**
Should install correctly as it did in version 24.0.x of docker

**Environment:**
 - Infrastructure: Openstack hypervisor, Debian 11 node
 - Deployment tool: manual

**Screenshots, Promtail config, or terminal output**
```latest: Pulling from grafana/loki-docker-driver
Digest: sha256:82270b494d8c157f035fde65bc0159076f4c4361b5bb061b4fc817e8db04234a
32dbe5a742d5: Complete 
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/etc/hosts" to rootfs at "/etc/hosts": mount /etc/hosts:/etc/hosts (via /proc/self/fd/6), flags: 0x5021: operation not permitted: unknown
```

**Comment by AnahiEstrell** [comment]
I have the same problem, were you able to solve it?

**Comment by JesseVS** [comment]
Not yet, it also happens after updating to v25 of docker rootless, so we cannot update docker at the moment.

**Comment by christoph-bessei** [comment]
We have the same issue, I think there's a related issue in the Docker/Moby repository:
https://github.com/moby/moby/issues/47248

**Comment by JesseVS** [comment]
The issue is fixed in docker version 26.0.0!
